ecommerce/programs/management/commands/migrate_program_offers_to_commercetools.py

```diff
@@ -0,0 +1,301 @@
+        discount_type = CT_CART_DISCOUNT_TYPE_MAP.get(offer.benefit.proxy().benefit_class_type)
+        discount_value = offer.benefit.value
+        type_value_key = f"{discount_type}-{discount_value}"
+
+        if type_value_key in discount_groups:
+            discount_groups[type_value_key]["program_uuids"].append(program_uuid)
+        else:
+            discount_groups[type_value_key] = {
+                "type": discount_type,
+                "value": discount_value,
+                "program_uuids": [program_uuid]
+            }
+
+    for _, value in discount_groups.items():
+        cart_discounts.append({
+            "type": value["type"],
+            "value": value["value"],
+            "program_uuids": value["program_uuids"]
+        })
+
+
+class Command(BaseCommand):
+    """Command to migrate program offers to Commercetools."""
+
+    def handle(self, *args, **options):
+        """Handle the command."""
+        try:
+            client = CommercetoolsAPIClient()
+        except CommandError as error:
+            logger.error(error)
+            return
+
+        sort_order = _get_highest_sort_order(client)
+
+        cart_discounts = []
+        _group_ten_percentage_offers(cart_discounts)
+        _group_other_offers(cart_discounts)
+
+        for discount_data in cart_discounts:
+            discount_type = discount_data["type"]
+            discount_value = discount_data["value"]
+            discount_value_in_cents = int(discount_value * 100)
+
+            logger.info(
+                "Checking existing cart discount with type %s and value %s in Commercetools.",
+                discount_type, discount_value
+            )
+            existing = client.get_cart_discounts_without_code_by_type_and_value(discount_type, discount_value_in_cents)
+            if not existing:
+                logger.info(
+                    "Failed to get discount with type %s and value %s. Cart discount not created.",
+                    discount_type, discount_value
+                )
+                continue
+
+            is_ten_percent_discount = (
+                discount_type == CT_PERCENTAGE_DISCOUNT_TYPE and discount_value_in_cents == TEN_PERCENT_DISCOUNT_IN_CENTS
+            )
+
+            if existing['count'] == 0:
+                sort_order += 0.00000000000001
+                logger.info(
+                    "Creating cart discount with type %s and value %s.",
+                    discount_type, discount_value
+                )
+                response = _create_cart_discount(
+                    client=client,
+                    discount_type=discount_type,
+                    discount_value=discount_value_in_cents,
+                    sort_order=sort_order,
+                    predicate=_create_target_predicate_from_program_uuids(
+                        discount_data["program_uuids"], is_ten_percent_discount
+                    )
+                )
```

Comment:
API calls (including this one) should be inside try-catch to handle unexpected exceptions.